### PR TITLE
Fix memory leaks in unit tests

### PR DESCRIPTION
**Pull request by JuergenReppSIT**
* Two leaks in  in swtpm unit test were fixed.
* Several leaks if the fapi json unit test were fixed.

Signed-off-by: Juergen Repp <juergen.repp@sit.fraunhofer.de>

### CODE REVIEW

File: test/unit/tcti-swtpm.c [hunk 1]
```diff
@@ -308,6 +308,9 @@ tcti_swtpm_init_fail_connect_test (void **state)
     will_return (__wrap_connect, -1);
     ret = Tss2_Tcti_Swtpm_Init (ctx, &tcti_size, "host=127.0.0.1,port=666");
     assert_int_equal (ret, TSS2_TCTI_RC_IO_ERROR);
+
+    free(((TSS2_TCTI_SWTPM_CONTEXT*)ctx)->conf_copy);
+    free(ctx);
```
Comment (williamcroberts) [inline test/unit/tcti-swtpm.c:313]: if it really matters and you want to guarantee your free gets called, you can setup the test with cmocka_unit_test_setup_teardown()
and the setup function set the void **s so *s points to the tcti that's calloc'd. Then the teardown will always get called and execute the free.

or you could just point the TCTI on the heap, a 4096 buffer should always be big enough, and just assert that size < sizeof(buf).

File: test/unit/fapi-json.c [hunk 2]
```diff
@@ -101,6 +101,79 @@ static void ifapi_cleanup_policy(TPMS_POLICY *policy)
...
+        SAFE_FREE(nv->event_log);
+    }
+}
+
+void
+cleanup_ifapi_object(
+    IFAPI_OBJECT * object)
+{
+    if (object != NULL) {
+        if (object->objectType != IFAPI_OBJ_NONE) {
```
Comment (tstruk) [inline test/unit/fapi-json.c:156]: Just a nit-pick - if you would combine the two if statements into one:
if (object != NULL && object->objectType != IFAPI_OBJ_NONE) {
...
it would save you one level of indentation.

**Comment by williamcroberts** [comment]
> -----Original Message-----
> From: Tadeusz Struk [mailto:notifications@github.com]
> Sent: Wednesday, May 20, 2020 1:28 PM
> To: tpm2-software/tpm2-tss <tpm2-tss@noreply.github.com>
> Cc: Roberts, William C <william.c.roberts@intel.com>; Comment
> <comment@noreply.github.com>
> Subject: Re: [tpm2-software/tpm2-tss] Fix memory leaks in unit tests (#1735)
> 
> @tstruk approved this pull request.
> 
> LGTM - I will hold on with merging in case you want to address Bill's feedback.

TL;DR you don't have to do it.

Just to be clear, that feedback is a nit. All the other test code assumes leaks on fail, so this one
doesn't need to be special. Extra points if it is. It also serves as historical reference
for anyone looking back in time that cmocka has this functionality and that’s how
you use it.

File: test/unit/tcti-swtpm.c (hunk 1 above)
Comment (JuergenReppSIT) [reply test/unit/tcti-swtpm.c:313]: In all other tests the ctx was passed via state and freed in the teardown method. Here ctx was allocated locally. So I thought it's OK to free it locally, to achieve running the test without failure if the address sanitizer is activated.
